jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -3226,6 +3249,10 @@ public String replaceAll(String regex, String substitute) {
 							helpers.putCharInArrayByIndex(newChars, newCharIndex++, replacement);
 						}
 					}
+					if ((replacement > 255) && (length == newCharIndex)) {
```

Comment:
`replacement > 255` implies that both the regex and the replacement are each 1 character long, in which case the length can't change which implies `length == newCharIndex`.
```java
					if (replacement > 255) {
```